hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -509,6 +509,256 @@ def check_core_dumps(reason=''):
+    users_uids_dirs = users_uids_dirs.split('\n') if users_uids_dirs != "" else []
+    error = []
+    for user_data in users_uids_dirs:
+        user_uid_dir = user_data.strip().split(" ")
+        if len(user_uid_dir) < 3:
+                user_uid_dir = user_uid_dir + ['']*(3-len(user_uid_dir))
+        if user_uid_dir[1].isdigit():
+            if int(user_uid_dir[1]) >= 1000 and os.path.isdir(user_uid_dir[2]) and user_uid_dir[0] is not "nfsnobody":
+                owner = _execute_shell_command("stat -L -c \"%U\" \"" + user_uid_dir[2] + "\"")
+                if owner is not user_uid_dir[0]:
+                    error += ["The home directory " + user_uid_dir[2] + " of user " + user_uid_dir[0] + " is owned by " + owner]
+        else:
+            error += ["User " + user_uid_dir[0] + " has invalid uid " + user_uid_dir[1]]
+
+    if error == []:
+        return True
+
+    return str(error)
+
+
+def check_users_dot_files(reason=''):
+    '''
+    Ensure users' dot files are not group or world writable
+    '''
+
+    users_dirs = _execute_shell_command("cat /etc/passwd | egrep -v '(root|halt|sync|shutdown)' | awk -F: '($7 != \"/sbin/nologin\") {print $1\" \"$6}'").strip()
+    users_dirs = users_dirs.split('\n') if users_dirs != "" else []
+    error = []
+    for user_dir in users_dirs:
+        user_dir = user_dir.split(" ")
+        if len(user_dir) < 2:
+                user_dir = user_dir + ['']*(2-len(user_dir))
+        if user_dir[1] is None or user_dir[1] == "":
+            error += ["User " + user_dir[0] + " does not have any home directory"]
+        else:
+            dot_files = _execute_shell_command("find " + user_dir[1] + " -name \".*\"").strip()
+            dot_files = dot_files.split('\n') if dot_files != "" else []
+            for dot_file in dot_files:
+                if os.path.isfile(dot_file):
+                    file_permission = _execute_shell_command("ls -ld " + dot_file + " | cut -f1 -d\" \"").strip()
+                    if file_permission[5] is "w":
+                        error += ["Group Write permission set on file " + dot_file + " for user " + user_dir[0]]
+                    if file_permission[8] is "w":
+                        error += ["Other Write permission set on file " + dot_file + " for user " + user_dir[0]]
+
+    if error == []:
+        return True
+
+    return str(error)
+
+
+def check_users_forward_files(reason=''):
+    '''
+    Ensure no users have .forward files
+    '''
+
+    users_dirs = _execute_shell_command("cat /etc/passwd | awk -F: '{ print $6 }'").strip()
+    users_dirs = users_dirs.split('\n') if users_dirs != "" else []
+    error = []
+    for user_dir in users_dirs:
+        if user_dir is not None and user_dir is not "" and os.path.isdir(user_dir):
+            forward_file = _execute_shell_command("find " + user_dir + " -name \".forward\"").strip()
+            if forward_file is not None and os.path.isfile(forward_file):
+                error += [forward_file]
```

Comment:
We need user details as well. In this user home dir, there is a .forward file exists. (something like that)